### PR TITLE
Adding a code of conduct

### PR DESCRIPTION
**Pull request by asmeurer**
This is based on the NumFOCUS code of conduct.

This will automatically apply to all repos in the Quansight organization. 

TODO: 

- [ ] Enforcement information
- [ ] Who needs to sign-off on this?
- [ ] Fill out all templated field in the CoC

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+Be kind to others. Do not insult or put down others. Behave professionally. Remember that harassment and sexist, racist, or exclusionary jokes are not appropriate for Quansight.
+
+All communication should be appropriate for a professional audience including people of many different backgrounds. Sexual language and imagery is not appropriate.
+
+Quansight is dedicated to providing a harassment-free community for everyone, regardless of gender, sexual orientation, gender identity and expression, disability, physical appearance, body size, race, or religion. We do not tolerate harassment of community members in any form.
+
+Thank you for helping make this a welcoming, friendly community for all.
+
+**CODE OF CONDUCT REPORTING FORM:** [*https://URL/to/your/CoC/reporting/form*](https://URL/to/your/CoC/reporting/form)
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:38]: This needs to be filled out.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+**All reports will be kept confidential.**
+
+In some cases we may determine that a public statement will need to be made. If that's the case, the identities of all involved parties and reporters will remain confidential unless those individuals instruct us otherwise.
+
+All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances.  The Quansight team commits to maintaining confidentiality with regard to the reporter of an incident.
+
+For possibly unintentional breaches of the code of conduct, you may want to respond to the person and point out this code of conduct (either in public or in private, whatever is most appropriate). If you would prefer not to do that, please report the issue to Quansight directly, or ask [PROJECT CoC Point Person(s)] for advice in confidence ([PROJECT CoC Point Person(s) Email address or other means of contact]). Complete contact information is below, under "How to Submit a Report."
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:108]: This needs to be filled out.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+Take care of each other. Alert Quansight if you notice a dangerous situation, someone in distress, or violations of this code of conduct, even if they seem inconsequential.
+
+#### How to Submit a Report
+
+**If you feel your safety is in jeopardy or the situation is an emergency, we urge you to contact local law enforcement before making a report to Quansight.** (In the U.S., dial 911.)
+
+Quansight is committed to promptly addressing any reported issues. If you have experienced or witnessed behavior that violates the Quansight Code of Conduct, please complete the form below to make a report.
+
+**REPORTING FORM:** [*https://URL/to/your/CoC/reporting/form*](https://URL/to/your/CoC/reporting/form)
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:118]: This need to be filled out.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+    -   +1 PHONE NUMBER (OPTIONAL)
+
+-   [NAME 3]
+
+    -   [*EMAIL*](mailto:EMAIL)
+
+    -   +1 PHONE NUMBER (OPTIONAL)
+
+You can reach the entire Code of Conduct Enforcement Team by emailing [***EMAIL ADDRESS***](mailto:EMAIL ADDRESS).
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:150]: This needs to be filled out.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+#### Conflicts of Interest
+
+In the event of any conflict of interest, the team member will immediately notify the Quansight Code of Conduct Enforcement Team and recuse themselves if necessary.
+
+If you are concerned about making a report that will be read by any of the
+above individuals, please reach out to [ADD ENFORCEMENT DETAILS HERE]
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:157]: This needs to be filled out.

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+### Enforcement: What Happens After a Report is Filed?
+
+What happens after a report is filed?
+
+#### Acknowledgment and Responding to Immediate Needs
+
+Quansight and/or our event staff will attempt to ensure your safety and help with any immediate needs, particularly at an in-person event.  Quansight will make every effort to **acknowledge receipt within 24 hours** (and we'll aim for much more quickly than that).
+
+<!-- PROJECT SHOULD REVIEW THE RESPONSE SCHEDULE LISTED ABOVE AND BELOW, AND DETERMINE WHETHER IT IS REALISTIC FOR THE PROJECT. -->
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:184]: This should be addressed.

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+
+We will contact the person who is the subject of the report to let them know what actions will be taken as a result of the report, if any.
+
+Our policy is to make sure that everyone aware of the initial incident is also made aware that official action has been taken, while still respecting the privacy of individuals. Quansight may choose to make a public report of the incident, while maintaining the anonymity of those involved.
+
+#### Appealing a Decision
+
+To appeal a decision of Quansight, contact [PROJECT CoC Point Person] via email at
+[*EMAIL ADDRESS*](mailto:EMAIL ADDRESS) with your appeal and
+the [PROJECT CoC TEAM/LEADERSHIP TEAM] will review the case.
```
Comment (asmeurer) [inline CODE_OF_CONDUCT.md:245]: This needs to be filled out.

**Comment by asmeurer** [comment]
Should we add some text here to clarify that this specifically applies to Quansight GitHub repos? I think there was some talk about a different code of conduct for internal use (unless we want to just use this same one there as well).

**Comment by dharhas** [comment]
I think even Quansight private GitHub repos could potentially have the same code of conduct.

**Comment by asmeurer** [comment]
If I am reading https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file correctly, the health files in this repo only automatically apply to public repos. Of course, it could still be Quansight policy that this applies to private repos as well. The "automatically applies" just means it automatically shows up as a link in places like the new issue form.

**Comment by dharhas** [comment]
Work in private repos is mainly by Quansight staff and they should already be covered by the company policy handbook.

**Comment by asmeurer** [comment]
My main motivation for adding this is to provide a code of conduct to handle situations that involve people who are not Quansight employees or clients, since we have several repos that are or aim to be community open source projects that have a wider participation than just Quansight directed uses. I don't know if any of this should be stated explicitly in the CoC. (also I want to reiterate that this is not motivated by any specific incident. No CoC violations have occurred to date in Quansight repos to my knowledge)

**Comment by kcpevey** [comment]
Closing in favor of #3

File: CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+- Sharing private content, such as emails sent privately or non-publicly, or direct message history, without the sender's consent
+- Continuing to initiate interaction (such as photography, recording, messaging, or conversation) with someone after being asked to stop
+- Sustained disruption of talks, events, or communications, such as heckling of a speaker
+- Publishing (or threatening to post) other people's personally identifying information ("doxing"), such as physical or electronic addresses, without explicit permission
+- Other unethical or unprofessional conduct
+- Advocating for, or encouraging, any of the above behaviors
+
+### Reporting Guidelines
+
+If you believe someone is violating the code of conduct, please report this in a timely manner. Code of conduct violations reduce the value of the community for everyone. The Quansight leadership team takes reports of misconduct very seriously and is committed to preserving and maintaining the welcoming nature of our community.
```
Comment (mattip) [inline CODE_OF_CONDUCT.md:100]: I think the order of these paragraphs is important. The one below "For possibly unintentional ..." should come first.

**Pull request closed without merging**